src/test/unit_tests/test_bucketspace.js

```diff
@@ -1193,3 +1192,138 @@ mocha.describe('nsfs account configurations', function() {
+    mocha.before(async () => fs_utils.create_fresh_path(full_path, 0o770));
+    mocha.after(async () => fs_utils.folder_delete(tmp_fs_root));
+
+    mocha.it('export dir as bucket', async function() {
+        await rpc_client.pool.create_namespace_resource({
+            name: nsr,
+            nsfs_config: {
+                fs_root_path: tmp_fs_root,
+                fs_backend: 'GPFS'
+            }
+        });
+        const obj_nsr = { resource: nsr, path: bucket_path };
+        await rpc_client.bucket.create_bucket({
+            name: bucket_name,
+            namespace: {
+                read_resources: [obj_nsr],
+                write_resource: obj_nsr
+            }
+        });
+
+        const policy = {
+            Version: '2012-10-17',
+            Statement: [{
+                Sid: 'id-1',
+                Effect: 'Allow',
+                Principal: { AWS: "*" },
+                Action: ['s3:PutBucketVersioning'],
+                Resource: [`arn:aws:s3:::${bucket_name}`]
+            }]
+        };
+        const { access_key, secret_key } = await generate_nsfs_account({ admin: true });
+        s3_admin = generate_s3_client(access_key, secret_key);
+        await s3_admin.putBucketPolicy({
+            Bucket: bucket_name,
+            Policy: JSON.stringify(policy)
+        }).promise();
+    });
+
+    mocha.it('create account uid, gid 5 - no permissions', async function() {
+        const { access_key, secret_key } = await generate_nsfs_account({ uid: 5, gid: 5 });
+        s3_uid5 = generate_s3_client(access_key, secret_key);
+    });
+
+    mocha.it('create account with uid, gid - default has permissions', async function() {
+        const { access_key, secret_key } = await generate_nsfs_account();
+        s3_uid6 = generate_s3_client(access_key, secret_key);
+    });
+
+    mocha.it('set bucket versioning - Enabled - should fail - no permissions', async function() {
+        try {
+            await s3_uid5.putBucketVersioning({ Bucket: bucket_name, VersioningConfiguration: { MFADelete: 'Disabled', Status: 'Enabled' } }).promise();
+            assert.fail(`put object succeeded for account without permissions`);
+        } catch (err) {
+            assert.ok(err.code === 'AccessDenied');
+        }
+    });
+
+    mocha.it('set bucket versioning - Enabled - admin - should fail - no permissions', async function() {
+        try {
+            await s3_admin.putBucketVersioning({ Bucket: bucket_name, VersioningConfiguration: { MFADelete: 'Disabled', Status: 'Enabled' } }).promise();
+            assert.fail(`put object succeeded for account without permissions`);
+        } catch (err) {
+            assert.ok(err.code === 'AccessDenied');
+        }
+    });
+
+    mocha.it('set bucket versioning - Enabled', async function() {
+        await s3_uid6.putBucketVersioning({ Bucket: bucket_name, VersioningConfiguration: { MFADelete: 'Disabled', Status: 'Enabled' } }).promise();
+    });
+
+});
```

Comment:
This test in general starting to be very long - maybe its time to split to test_bucketspace_versionning?

Reply:
moved it out to a new test file